VO Tester/runner.py

```diff
@@ -64,41 +78,46 @@ def run(args):
 
   print("Browser is: {}".format(args.browser))
   browser.open_new_tab(location)
-  time.sleep(2)
 
   # start voiceover
-  os.system("osascript -e 'tell application \"System Events\" to key code 96 using command down'")
+  pyautogui.hotkey("command", "f5")
 
   # start server here to avoid (some) startup noise
   print("Starting server....")
   httpd = HTTPServer(('localhost', 8080), RequestHandler)
-  httpd.timeout = 5
+  httpd.timeout = 1
   print("Server started...")
 
+  time.sleep(2)
   with open(args.inputFile) as f:
     for line in f:
       line = line[:-1]
+      print(f"# {line}")
       global currTime
       currTime = time.time()
       if (line == "next"):
-        os.system("osascript -e 'tell application \"System Events\" to key code 124 using {option down, control down}'")
+        pyautogui.hotkey("option", "ctrl","right")
       elif (line == "prev"):
-        os.system("osascript -e 'tell application \"System Events\" to key code 123 using {option down, control down}'")
+        pyautogui.hotkey("option", "ctrl", "left")
       elif (line == "in"):
-        os.system("osascript -e 'tell application \"System Events\" to key code 125 using {option down, control down, shift down}'")
+        pyautogui.hotkey("shift", "option", "ctrl", "down")
       elif (line == "out"):
-        os.system("osascript -e 'tell application \"System Events\" to key code 126 using {option down, control down, shift down}'")
+        pyautogui.hotkey("shift", "option", "ctrl", "up")
       elif (line == "activate"):
-        os.system("osascript -e 'tell application \"System Events\" to key code 49 using {option down, control down}'")
+        pyautogui.hotkey("option", "ctrl", "space")
       else:
         print("Error parsing command: {}. Skipping".format(line))
 
-      while (httpd.handle_request()):
-        continue
+      global responded
+      responded = False
+      for i in range(5):
+        httpd.handle_request()
+        if responded:
```

Comment:
Isn't this going to break after every text string (where the delta is > 0)? I thought part of the point was that we might get multiple strings returned from one command? Otherwise, why not just call `handle_request()` outside of the while loop?

Reply:
it will break after the first text string with a positive delta (ie. the utterance VO dispatches right after the given command). if there are any followup utterances they will be found in the next loop iteration but hopefully thrown out because their delta will be smaller than 0 (ie. that utterance was sent before the next vo command was given).

Reply:
so we're assuming only the first utterance is important for performance and correctness, then? 
I think that feels right, not sure how many strings after the first are content related vs. voiceover trying to be helpful with things like "to activate this link, press..."

Reply:
yeah. i think that can be useful in the future too. not sure how it fits in the current format/tool